nbinteract/plotting.py

```diff
@@ -619,3 +638,26 @@ def _array_or_placeholder(
     if isinstance(maybe_iterable, collections.Iterable):
         return np.array([i for i in maybe_iterable])
     return placeholder
+
+##############################################################################
+# Figure Class
+##############################################################################
+class Figure:
+    def __init__(self, options={}):
+        self.options = options
+        self.figure = figure(options=options)
+        self.widget_lst = []
+
+
+    def scatter(self, x_fn, y_fn, *, options={}, **interact_params):
+        box = scatter(x_fn, y_fn, fig=self.figure,
+                                 options=options, **interact_params)
+        widget = box.children[0]
+        fig = box.children[1]
+        self.widget_lst.append(widget)
+
+    def show(self):
+        for widget in self.widget_lst:
+            display(widget)
+        display(self.figure)
```

Comment:
Let's define a `_ipython_display_` function on the `Figure` class which calls the `show` method. This way, we won't have to call `.show()` when the figure is the last line of a cell and we can make all plotting functions return a Figure. Previous plotting calls like 

```python
nbi.scatter(...)
```

will still display a figure.

Reply:
Fixed to return Vbox which will show if it is the last statement ran